### PR TITLE
Ensure the discovery endpoint is validated and all required fields ar…

### PR DESCRIPTION
**Pull request by nc-fkl**
Ensure the discovery endpoint is validated and all required fields are verified

### CODE REVIEW

File: lib/Controller/SettingsController.php [hunk 1]
```diff
@@ -63,12 +63,59 @@ public function __construct(
 		$this->crypto = $crypto;
 	}
 
+	public function isDiscoveryEndpointValid($url) {
+		// Initialize cURL session
```
Comment (juliushaertl) [inline lib/Controller/SettingsController.php:67]: This should use https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/http_client.html, not plain curl

File: lib/Controller/SettingsController.php [hunk 2]
```diff
@@ -47,20 +49,63 @@ class SettingsController extends Controller {
...
+				
+				foreach ($requiredFields as $field) {
+					if (!isset($data[$field])) {
+						$result['missingFields'][] = $field;
+					}
+				}
+			} else {
+				// Set isReachable to false if http code wasn't 200
+				$result['isReachable'] = false;
+			}
```
Comment (juliushaertl) [inline lib/Controller/SettingsController.php:103]:
```suggestion
			}
```

No need for this else branch as isReachable is only set to true if the status code is 200 in line 86

File: lib/Controller/SettingsController.php [hunk 3]
```diff
@@ -47,20 +49,63 @@ class SettingsController extends Controller {
...
+		} catch (Exception $e) {
+			$result['isReachable'] = false;
```
Comment (juliushaertl) [inline lib/Controller/SettingsController.php:105]: Same here, but still worth to catch and best log the exception here

**Review by juliushaertl — COMMENTED** [review]
Two additional minor comments, otherwise this looks sane :)

File: lib/Controller/SettingsController.php [hunk 4]
```diff
@@ -47,20 +50,64 @@ class SettingsController extends Controller {
...
+		];
+
+		try {
+			$client = $this->clientService->newClient();
+			$response = $client->get($url);
+			$httpCode = $response->getStatusCode();
+			$body = $response->getBody();
+
+			// Check if the request was successful
+			if ($httpCode == 200 && !empty($body)) {
```
Comment (julien-nc) [inline lib/Controller/SettingsController.php:90]:
```suggestion
			if ($httpCode === Http::STATUS_OK && !empty($body)) {
```
Use triple equal and use `OCP\AppFramework\Http` constants for HTTP response codes.

File: lib/Controller/SettingsController.php [hunk 5]
```diff
@@ -47,20 +50,64 @@ class SettingsController extends Controller {
...
+				];
+				
+				foreach ($requiredFields as $field) {
+					if (!isset($data[$field])) {
+						$result['missingFields'][] = $field;
+					}
+				}
+			}
+		} catch (Exception $e) {
+			$this->logger->error('Error - discovery endpoint validation: ' . $e);
```
Comment (julien-nc) [inline lib/Controller/SettingsController.php:107]:
```suggestion
			$this->logger->error('Discovery endpoint validation error', ['exception' => $e]);
```
So we get more info on the exception.

**Pull request merged**